pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -530,6 +531,31 @@ func (s *discovery) cleanCache(alivePids map[int32]struct{}) {
 	}
 }
 
+// updateServicesCPUStats updates the CPU stats of the service, as well as the
+// global CPU time cache for future updates.
+func (s *discovery) updateServicesCPUStats() error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+
+	if time.Since(s.lastCPUTimeUpdate) < s.config.cpuUsageUpdateDelay {
+		return nil
+	}
+
+	globalCPUTime, err := getGlobalCPUTime()
+	if err != nil {
+		return fmt.Errorf("could not get global CPU time: %w", err)
+	}
+
+	for pid, serviceInfo := range s.cache {
```

Comment:
Shouldn't this updating the stats in the list of services we are returning now (passed in from the caller) instead of only in the cache? Otherwise the computed CPU usage is only reported the next time.